addons/l10n_in/models/account_move_line.py

```diff
@@ -38,3 +63,210 @@ def _l10n_in_check_invalid_hsn_code(self):
+                'l10n_in.tax_tag_base_cgst_rc', 'l10n_in.tax_tag_cgst_rc',
+                'l10n_in.tax_tag_base_igst_rc', 'l10n_in.tax_tag_igst_rc',
+                'l10n_in.tax_tag_base_cess_rc', 'l10n_in.tax_tag_cess_rc'
+            ),
+            'gst': get_tag_ids(
+                'l10n_in.tax_tag_base_sgst', 'l10n_in.tax_tag_sgst',
+                'l10n_in.tax_tag_base_cgst', 'l10n_in.tax_tag_cgst',
+                'l10n_in.tax_tag_base_igst', 'l10n_in.tax_tag_igst',
+                'l10n_in.tax_tag_base_cess', 'l10n_in.tax_tag_cess'
+            ),
+            'nil': get_tag_ids(
+                'l10n_in.tax_tag_exempt', 'l10n_in.tax_tag_nil_rated', 'l10n_in.tax_tag_non_gst_supplies'
+            ),
+            'export': get_tag_ids(
+                'l10n_in.tax_tag_zero_rated', 'l10n_in.tax_tag_base_igst', 'l10n_in.tax_tag_igst',
+                'l10n_in.tax_tag_base_cess', 'l10n_in.tax_tag_cess'
+            ),
+            'igst_lut': get_tag_ids('l10n_in.tax_tag_igst_lut')
+        }
+
+    @api.depends('move_id.l10n_in_gst_treatment', 'move_id.amount_total', 'move_id.l10n_in_state_id', 'company_id', 'tax_tag_ids')
+    def _compute_l10n_in_gstr_section(self):
+
+        def has_tags(category):
+            return any(tag in tax_tags for tag in tax_tags_ids[category])
+
+        def is_invoice_or_bill(move):
+            return (
+                (move.move_type in ('out_invoice', 'out_receipt', 'in_invoice', 'in_receipt'))
+                and not move.debit_origin_id
+            )
+
+        def get_transaction_type(move):
+            return 'intra_state' if move.l10n_in_state_id == move.company_id.state_id else 'inter_state'
+
+        def is_b2b_rcm(line):
+            return (
+                is_invoice_or_bill(line.move_id) and
+                line.move_id.l10n_in_gst_treatment in ('regular', 'composition', 'uin_holders') and
+                has_tags('gst_rc')
+            )
+
+        def is_b2b_regular(line):
+            return (
+                is_invoice_or_bill(line.move_id) and
+                line.move_id.l10n_in_gst_treatment in ('regular', 'composition', 'uin_holders') and
+                has_tags('gst')
+            )
+
+        def is_b2cl(line):
+            return (
+                is_invoice_or_bill(line.move_id) and
+                line.move_id.l10n_in_gst_treatment in ('unregistered', 'consumer') and
+                has_tags('gst') and
+                get_transaction_type(line.move_id) == 'inter_state' and
+                line.move_id.amount_total > 100000
+            )
+
+        def is_b2cs(line):
+            return (
+                line.move_id.l10n_in_gst_treatment in ('unregistered', 'consumer') and
+                has_tags('gst') and
+                (get_transaction_type(line.move_id) == 'intra_state' or
+                (get_transaction_type(line.move_id) == 'inter_state' and
+                    (is_invoice_or_bill(line.move_id) and line.move_id.amount_total <= 100000) or
```

Comment:
You need to update this as per this commit https://github.com/odoo/enterprise/commit/f33e047ca8433b6875a8e88b4f686fa9be6a9985